core/journal/journal.go

```diff
@@ -58,13 +61,31 @@ func (this *Journal) NewEntry(request *http.Request, response *http.Response, mo
 		this.entries = append(this.entries[:0], this.entries[1:]...)
 	}
 
-	this.entries = append(this.entries, JournalEntry{
+	entry := JournalEntry{
 		Request:     &payloadRequest,
 		Response:    payloadResponse,
 		Mode:        mode,
 		TimeStarted: started,
 		Latency:     time.Since(started),
-	})
+	}
+
+	this.entries = append(this.entries, entry)
+
+	buf := new(bytes.Buffer)
+	enc := json.NewEncoder(buf)
+	// do not escape characters in HTML like < and >
+	enc.SetEscapeHTML(false)
+	err := enc.Encode(convertJournalEntry(entry))
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error": err.Error(),
+		}).Error("invalid journal entry")
+	} else {
+		log.WithFields(log.Fields{
+			"json": buf.String(),
+		}).Debug("journal entry")
+	}
+
```

Comment:
can we do the json marshalling only when debug is enabled? For large request/response body, it will significantly impact the performance which we learned from the past.

Reply:
Thank you. I add a check by `logrus.IsLevelEnabled(logrus.DebugLevel)` which means the level is greater than or equal to DEBUG.